crates/typst-cli/src/compile.rs

```diff
@@ -198,6 +198,41 @@ enum ImageExportFormat {
     Svg,
 }
 
+struct OutputTemplate;
+
+impl OutputTemplate {
+    const INDEXABLE: [&'static str; 3] = ["{p}", "{0p}", "{n}"];
+
+    fn has_indexable_template(output: &str) -> bool {
+        Self::INDEXABLE.iter().any(|template| output.contains(template))
+    }
+
+    fn indexable_template() -> [&'static str; 3] {
+        Self::INDEXABLE
+    }
```

Comment:
why a method that just forwards to the const?

Reply:
good point